api/proto/teleport/legacy/types/events/events.proto

```diff
@@ -7950,3 +7953,97 @@ message UserLoginAccessListInvalid {
     (gogoproto.jsontag) = ""
   ];
 }
+
+// AWSICResourceSyncEvent defines import and reconciliation event data for a
+// account, account assignment, permission set or user group type.
+message AWSICResourceSyncEvent {
+  //  Resource represents a single user group, account, account assignment or permission set.
+  message Resource {
+    // ID is an ID of the resource.
+    string id = 1 [(gogoproto.jsontag) = "id, omitempty"];
+    // Name is a display name of the resource.
+    string name = 2 [(gogoproto.jsontag) = "name, omitempty"];
+    // ARN is the ARN of the resource.
+    string arn = 3 [(gogoproto.jsontag) = "arn,omitempty"];
+    // AssignedPermissionSetARN holds the ARN value of the assigned permission set.
+    // Only used for an account assignment resource.
+    string assignedPermissionSetARN = 4 [(gogoproto.jsontag) = "assigned_permission_set_arn, omitempty"];
```

Comment:
"Only used for account assignment resource" tells me this data model might not be the best fit, especially if we ever add more types or want more unique fields. You might consider using a `oneof`